src/inmanta/ast/__init__.py

```diff
@@ -67,10 +67,18 @@ def __str__(self) -> str:
 class Locatable(object):
 
     def __init__(self):
-        self.location = None  # type: Location
+        self._location = None  # type: Location
+
+    def set_location(self, location: Location):
+        assert location is not None
+        assert location.file == "internal" or location.lnr > 0
```

Comment:
perhaps modify the "internal" lines to also be >0, that number is arbitrary anyway 
this code is executed quite often so the additional string compare can be avoided

Reply:
Could be useful, not sure yet if we keep these asserts or not